python/ray/services.py

```diff
@@ -580,6 +592,35 @@ def start_reaper():
     Returns:
         ProcessInfo for the process that was started.
     """
+    if sys.platform.startswith("linux"):
+        # No need for a reaper on Linux.
+        # The kernel supports fate-sharing natively, so just initilize it.
+        global libc
+        if libc is None:
+            try:
+                libc = ctypes.CDLL(None)
+            except TypeError:
+                libc = False
+        global prctl
+        if prctl is None and libc is not None:
+            try:
+                prctl = libc.prctl
+                prctl.restype = ctypes.c_int
+                prctl.argtypes = [
+                    ctypes.c_int, ctypes.c_ulong, ctypes.c_ulong,
+                    ctypes.c_ulong, ctypes.c_ulong
+                ]
+            except AttributeError:
+                prctl = False
+        if prctl:
+            global set_psigdeath
```

Comment:
Can we just do these checks inside of `preexec_fn()` instead of using this global variable function as a flag?

Reply:
Not really; at that point it's more or less too late to know if we should've started a reaper or not. This lets us bail out and use the reaper in case any of these checks fail, before we actually spawn processes.

Also, `preexec_fn` is a delicate environment since it's in a fork context; you want to minimize the logic inside it, so it seemed best not to try to something like loading a shared library there.